test/gtest/onehot.hpp

```diff
@@ -0,0 +1,169 @@
+    }
+
+    std::vector<size_t> GetInput()
+    {
+        if((N != 0) && (C != 0) && (D != 0) && (H != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, D, H, W});
+        }
+        else if((N != 0) && (C != 0) && (H != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, H, W});
+        }
+        else if((N != 0) && (C != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, W});
+        }
+        else if((N != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, W});
+        }
+        else if((N != 0))
+        {
+            return std::vector<size_t>({N});
+        }
+        else
+        {
+            std::cout << "Error Input Tensor Lengths\n" << std::endl;
+            return std::vector<size_t>({0});
+        }
+    }
+};
+
+std::vector<OneHotTestCase> OneHotTestConfigs()
+{ // n c d h w
+    return {
+        {1, 0, 0, 0, 10, 10, 1},
+        {2, 0, 0, 10, 10, 10, 1},
+        {4, 1, 0, 100, 100, 100, 1},
+        {8, 3, 1, 20, 100, 200, 1},
+        {16, 3, 1, 100, 100, 200, 1},
+        {1, 1, 1, 1, 5000, 5000, 1},
+    };
+}
+
+template <typename T = int>
+struct OneHotTest : public ::testing::TestWithParam<OneHotTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle  = get_handle();
+        onehot_config  = GetParam();
+        int gen_range  = onehot_config.num_classes;
+        auto gen_value = [&gen_range](auto...) { return prng::gen_A_to_B(0, gen_range - 1); };
+
+        auto in_dims = onehot_config.GetInput();
+
+        input = tensor<T>{in_dims}.generate(gen_value);
+
+        std::vector<size_t> out_dims;
+
+        for(unsigned long in_dim : in_dims)
+        {
+            out_dims.push_back(in_dim);
+            onehot_config.input_size *= in_dim;
+        }
+
+        out_dims.push_back(onehot_config.num_classes);
+
+        output = tensor<T>{out_dims};
+        std::fill(output.begin(), output.end(), 0);
+
+        ref_output = tensor<T>{out_dims};
+        std::fill(ref_output.begin(), ref_output.end(), 0);
+
+        input_dev  = handle.Write(input.data);
+        output_dev = handle.Write(output.data);
+    }
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+
+        miopenStatus_t status;
+
+        status = miopen::OneHot(handle,
+                                input.desc,
+                                input_dev.get(),
+                                onehot_config.input_size,
+                                output.desc,
+                                output_dev.get(),
+                                onehot_config.num_classes);
+
+        EXPECT_EQ(status, miopenStatusSuccess);
+        cpu_onehot<T>(input, ref_output, onehot_config.input_size, onehot_config.num_classes);
+
+        output.data = handle.Read<T>(output_dev, output.data.size());
+    }
+
+    void Verify()
+    {
+        double threshold = 1e-6;
```

Comment:
Tolerance is used to compare results of floats. For int, please check if the values are the same.